### ISSUE TITLE
"Synchronous conditional normalization not supported sync normalizing"

### ISSUE THREAD

**Issue by adelespinasse** [jspm/jspm-cli#2367]
**JSPM Version:** 0.17.0-beta.45

**Transpiler Plugin(s):** babel

**Details:**
This error occurs intermittently when I compile a bundle with `jspm build --node`. I'm not sure what percentage of the time it happens, but it's very odd that it seems to be random. I can literally run the exact same command several times with no changes, and get the error on some runs and not others. If it was consistent, I might think it was a bug in the code being bundled, but as is I think it's probably jspm/SystemJS.

I have no idea what the error message means. Other online references to it don't seem related.

The full output is:
```
     Creating the single-file build for server/services/filters/main.js...

err  Error: Synchronous conditional normalization not supported sync normalizing ./node.js in file:///C:/Users/aldel/agora/playdoh/public/jspm_packages/npm/debug@2.6.8
    at doMapSync (C:\Users\aldel\agora\playdoh\node_modules\systemjs-builder\node_modules\systemjs\dist\system.src.js:2304:13)
    at applyPackageConfigSync (C:\Users\aldel\agora\playdoh\node_modules\systemjs-builder\node_modules\systemjs\dist\system.src.js:2275:22)
    at SystemJSNodeLoader.<anonymous> (C:\Users\aldel\agora\playdoh\node_modules\systemjs-builder\node_modules\systemjs\dist\system.src.js:2505:14)
    at SystemJSNodeLoader.normalizeSync (C:\Users\aldel\agora\playdoh\node_modules\systemjs-builder\node_modules\systemjs\dist\system.src.js:4459:30)
    at SystemJSNodeLoader.loader.normalizeSync (C:\Users\aldel\agora\playdoh\node_modules\systemjs-builder\lib\builder.js:228:42)
    at toPackagePath (C:\Users\aldel\agora\playdoh\node_modules\systemjs-builder\lib\trace.js:236:35)
    at C:\Users\aldel\agora\playdoh\node_modules\systemjs-builder\lib\trace.js:300:20
```
I think it's always on package `npm:debug` when it crashes. It happens at least in versions 2.2.0 and 2.6.8 of that package.

 I first noticed this in jspm 0.17.0-beta.32, and for a while I thought that later versions didn't have this problem, but now I'm getting it in 0.17.0-beta.45.

I have not seen it happen when building for browsers (without --node flag), but I'm not sure if I have a dependency on npm:debug in those bundles.

I can try to create a minimal reproduction if that would be helpful. I'd start by making a simple project that does nothing but depend on npm:debug, but I'm not sure if that would do it.

**Comment by guybedford** [comment]
@adelespinasse if you can share a replication that would be great. That's really odd that it only happens sporadically.

**Comment by adelespinasse** [comment]
No luck so far on a simple replication. I tried simply importing the debug package, and also importing and using one that uses it (express).

Is there any way to get more information about what is happening when the error occurs? The error message says what package it was working on, but nothing about what other package was importing it, for example. I found the `--log` option, but the default already seems to be the most verbose mode.

**Comment by adelespinasse** [comment]
Got it! Here's a repo: https://github.com/adelespinasse/jspm-syncnorm-bug

- Clone the repo
- `npm install`
- `jspm install`
- `jspm build jspm-syncnorm-bug --node`

I just added all the packages in my big project that depend on `npm:debug`. Still don't know which one causes the problem.

I get the error pretty frequently. (I hope this isn't a Windows-only bug or something.)

**Comment by guybedford** [comment]
Fixed in https://github.com/systemjs/builder/commit/19b73d499bc7723637da6470b7ed4b79aa8f0ef9.

**Comment by guybedford** [comment]
Released in 0.17.0-beta.46.
